docs/guides/understand_logs_and_metrics.md

```diff
@@ -0,0 +1,252 @@
+
+- Here we have `TFLOP/s/device: 31.496`. Let's try to verify manually: $31.86 /1.012 = 31.482$. Not exactly same but close, since the both tflop and time are rounded in log.
+- Further, we can calculate **Model Flop Utilization (MFU)** from this:
+  
+$$\text{MFU} = \frac{\text{tflop/s/device}}{\text{peak hardware tflop/s}}$$
+  
+  For TPU v5p, $\text{peak hardware tflop/s}=459$. Thus, $31.496 / 459 = 6.86\%$.
+
+**Tokens Per Second Per Device (throughput)**
+
+-  It is [computed](https://github.com/AI-Hypercomputer/maxtext/blob/e969faabbb571285a51545530f34d8f0a9f237e9/MaxText/metric_logger.py#L197-L199) as
+
+$$\text{token/s/device} = \frac{\text{number of tokens per device}}{\text{measured step time in seconds}}$$
+
+  - The numerator is from [calculate_tokens_training_per_device](https://github.com/AI-Hypercomputer/maxtext/blob/e969faabbb571285a51545530f34d8f0a9f237e9/MaxText/maxtext_utils.py#L151)
+
+$$\text{number of tokens per device} = \text{per device batch size} \times \text{max target length}$$
+
+  - Here we have `Tokens/s/device: 2024.558`. Let's try to verify manually: $1 \times 2048 /1.012 = 2023.715$. Not exactly same but close, since the time is rounded in log.
+
+
+### 4.2 Learning Metrics
+
+**Loss**. The loss is the key indicator of learning progress, which should decrease over time steps. Ideally, we want it to converge to a small value.
+
+**Total Weights**. When discussing the throughput, we have $\text{number of tokens} = \text{per device batch size} \times \text{max target length} \times \text{number of device}$. In data preprocessing, for each sentence, we truncate or pad to max target length. The pad tokens are meaningless and the loss are calculated based on the nonpad tokens. Thus, we monitor $\text{number of nonpad tokens}$, which is shown as [total weights](https://github.com/AI-Hypercomputer/maxtext/blob/f82ce194c490d668b14574a072a0a630c27bbd6e/MaxText/train.py#L307).
+- Here we see `total_weights: 8192` for all steps. This is because we are using `dataset_type=synthetic`, where all sentences are generated with length of `max_target_length=2048`. As a result, there are no padded tokens and total weights = number of tokens. 
+- However, in real dataset, sentences can have variable lengths and total weights < number of tokens. For example, we can set `dataset_type=tfds dataset_path=gs://maxtext-dataset dataset_name='c4/en:3.0.1'`, and will see total weights is smaller than 8192:
+  ```
+  completed step: 8, seconds: 0.983, TFLOP/s/device: 32.418, Tokens/s/device: 2083.764, total_weights: 7805, loss: 9.607
+  completed step: 9, seconds: 0.983, TFLOP/s/device: 32.397, Tokens/s/device: 2082.441, total_weights: 7100, loss: 9.794
+  ```
+- For better convergence, we want to have large total weights. For example, MaxText allows for [packing](https://github.com/AI-Hypercomputer/maxtext/blob/f82ce194c490d668b14574a072a0a630c27bbd6e/MaxText/sequence_packing.py#L39) multiple short sequences into one.
```

Comment:
Shall we point to this config? https://github.com/AI-Hypercomputer/maxtext/blob/eff346c028092c4f4fd421e5c5343308def5de5a/MaxText/configs/base.yml#L454